services/src/messaging/src/main/java/org/openkilda/messaging/model/SwitchId.java

```diff
@@ -0,0 +1,132 @@
+/* Copyright 2017 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.messaging.model;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * Represents a switch id entity.
+ */
+public class SwitchId implements Serializable, Comparable<SwitchId> {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Switch switchId.
+     */
+    private final long switchId;
+
+    /**
+     * Instance constructor.
+     *
+     * @param switchId switch id.
+     */
+    public SwitchId(String switchId) {
+
+        String[] bytesString = switchId.split("[-:]");
+
+        long value = 0;
+        for (String oneByte : bytesString) {
+            value = (value << 8) + ((byte) Integer.parseInt(oneByte, 16) & 0xff);
+        }
+
+        this.switchId = value;
+    }
+
+    /**
+     * Return the MAC address of switch.
+     *
+     * @return the MAC address of switch.
+     */
+    public String extractMacAddress() {
```

Comment:
`extract` is incorrect term. You return one of possible representation of StringId value. You will need to have set of this representation methods...

* `asMacAddress()`
* `asLong()`
* `asSomethingElse()`
...

Someone prefer `toMacAddress` variant... but it definetely not `extractSomething`.